entrypoint.sh

```diff
@@ -26,6 +26,35 @@ if ! whoami &> /dev/null; then
     echo "${USER_NAME:-user}:x:$(id -u):" >> /etc/group
   fi
 fi
+
+preventWorkspaceStop() {
+  # Ensure that main environment variable are configured
+  if [ -z "${CHE_API_INTERNAL}" ]; then
+      echo "CHE_API_INTERNAL is not configured in environment."
+      exit 1
+  fi
+  if [ -z "${CHE_MACHINE_TOKEN}" ]; then
+      echo "CHE_MACHINE_TOKEN is not configured in environment."
+      exit 1
+  fi
+  if [ -z "${CHE_WORKSPACE_ID}" ]; then
+      echo "CHE_WORKSPACE_ID is not configured in environment."
+      exit 1
+  fi
+
+  # Todo(vzhukovs): check for client certificates that might be included into request in case if script runs on minikube with self-signed certificates
+
+  # Start making requests to avoid idling, make a request to activity tracker each minute
+  while true; do
+      curl -X PUT \
+      -H "Authorization: Bearer ${CHE_MACHINE_TOKEN}" \
+      "${CHE_API_INTERNAL}/activity/${CHE_WORKSPACE_ID}"
+
+      sleep 1m
+  done
+}
+
 cat /etc/passwd
 id
+preventWorkspaceStop &
```

Comment:
I would move all this logic to a bash script and add it into supervisord (which spawns all daemon)

Reply:
Moved to dedicate bash script